docs: Add common use cases (#4677)
CloudFire-LLC/cloudfire: website/src/app/kb/use-cases/nat-gateway/readme.mdx

import SupportOptions from "@/components/SupportOptions";
import Link from "next/link";
import Alert from "@/components/DocsAlert";
import Image from "next/image";
import PlanBadge from "@/components/PlanBadge";

<PlanBadge plans={["starter", "team", "enterprise"]}>

# Route Traffic Through a Public IP

</PlanBadge>

In this guide, we'll walk through the configuration required to use Firezone to
route traffic through a public IP address, sometimes known as a NAT Gateway
configuration.

> This is commonly used when you need your team's traffic to appear to come from a

Reviewer: ```suggestion
Use this when you need your team's traffic to appear to come from a
```

single, static IP address to use services that require source traffic to come

> from specified IP whitelists.

Reviewer: ```suggestion
from specified IP allowlists.
```
To match Github's precedent and other places in our docs

Author: 👍

After completing this guide, your team's traffic will be routed to a Firezone
Gateway and then out to the internet using its public IP address.

<Alert color="info">
  See our our [Terraform
  examples](https://www.github.com/firezone/firezone/tree/main/terrraform/examples)

>   for an high availability example of this guide using Terraform on Google Cloud

Reviewer: ```suggestion
  for a high availability example of this guide using Terraform on Google Cloud
```

  Platform.
</Alert>

## Prerequisites

- A Site reserved for this use case. [Create a Site](/kb/deploy/sites) if you
  haven't already.
- One or more Gateways deployed within the Site.
  [Deploy a Gateway](/kb/deploy/gateways) if you haven't done so yet.

<Alert color="warning">
  This guide assumes the Gateway has a public IP attached. This is required to
  function as a NAT gateway.
</Alert>

## Step 1: Create Resource(s) matching the traffic you want to route

1. In your [admin portal](https://app.firezone.dev), go to `Sites -> <site>` and
   click the `Add Resource` button.

> 1. Add Resource(s) for each service that requires an IP whitelist. For example,

Reviewer: ```suggestion
1. Add Resource(s) for each service that requires an IP allowlist. For example,
```
